.github/workflows/os_test.yml

```diff
@@ -2,45 +2,13 @@ name: course_os test suite
 
 on: [push]
 
-defaults:
-  run:
-     working-directory: src
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
 
-      - name: Cache Toolchain
-        id: cache-toolchain
-        uses: actions/cache@v1
-        with:
-            path: src/toolchain
-            key: toolchain
-
-      - name: Cache Qemu
-        id: cache-qemu
-        uses: actions/cache@v1
-        with:
-            path: src/qemu/qemu
-            key: qemu
-
-      - name: Install Dependencies
-        run: |
-          sudo apt-get update &&
-          sudo apt-get install build-essential python3 wget texinfo zlib1g-dev ninja-build -y
-
-      - name: Build Toolchain
-        if: steps.cache-toolchain.outputs.cache-hit != 'true'
-        run: rm -rf toolchain/arm-none-eabi/ && make toolchain
-
-      - name: Build Qemu
-        if: steps.cache-qemu.outputs.cache-hit != 'true'
-        run: rm -rf qemu/qemu* && make qemu
-
-      - name: Compile
-        run: make build
-
-      - name: Test
-        run: make test
+      - run: |
+          docker build -t course_os .
+          docker run course_os make test
```

Comment:
don't you need to install docker as a dependency? I don't think ubuntu comes with it by default

Reply:
I would think so as well, but the Github CI docs specify that the image indeed includes it.

Reply:
oh sure, that's fine